packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -265,12 +266,17 @@ export function convertSnapshotTreeToSummaryTree(
         0x19e /* "There should not be commit tree entries in snapshot" */);
 
     const builder = new SummaryTreeBuilder();
-    for (const [key, value] of Object.entries(snapshot.blobs)) {
-        // The entries in blobs are supposed to be blobPath -> blobId and blobId -> blobValue
-        // and we want to push blobPath to blobValue in tree entries.
-        if (snapshot.blobs[value] !== undefined) {
-            const decoded = fromBase64ToUtf8(snapshot.blobs[value]);
-            builder.addBlob(key, decoded);
+    for (const [path, id] of Object.entries(snapshot.blobs)) {
+        // 0.44 back-compat We still put contents in same blob for back-compat so need to add blob
+        // only for blobPath -> blobId mapping and not for blobId -> blob value contents.
+        if (snapshot.blobs[id] !== undefined) {
+            let decoded: string;
+            if ((snapshot as any).blobsContents !== undefined) {
+                decoded = bufferToString((snapshot as any).blobsContents[id], "utf8");
```

Comment:
I believe summaries support binary blobs, so we should avoid conversion if possible. That said, I'm not sure how well this path is tested, as it's likely not exercised,

Reply:
This is only exercised while creation of summary for attach when that container was previously rehydrated from a snapshot.

Reply:
So we can just pass the buffer here?

Reply:
This path is tested. We also have the capability in webpack-fluid-loader and also have test cases too.